File: 1910-check-if-binary-string-has-at-most-one-segment-of-ones/check-if-binary-string-has-at-most-one-segment-of-ones.py
class Solution(object):
    def checkOnesSegment(self, s):
        count = 0
        i = 0
        while i < len(s):
            if s[i] == '1':
                count += 1
                if count > 1:
                    return False
                # Skip entire segment of 1s
                while i < len(s) and s[i] == '1':
                    i += 1
            else:
                i += 1
        return True

File: 1910-check-if-binary-string-has-at-most-one-segment-of-ones/check-if-binary-string-has-at-most-one-segment-of-ones.c
#include <string.h>
#include <stdbool.h>

bool checkOnesSegment(char* s) {
   
    for (int i = 0; s[i + 1] != '\0'; i++) {
        if (s[i] == '0' && s[i + 1] == '1') {
            return false;
        }
    }
    return true;
}